packages/lib/server/getLuckyUser.ts

```diff
@@ -291,6 +331,223 @@ async function filterUsersBasedOnWeights<
   return remainingUsersAfterWeightFilter;
 }
 
+type AttributeWithWeights = {
+  name: string;
+  slug: string;
+  type: AttributeType;
+  id: string;
+  options: {
+    id: string;
+    value: string;
+    slug: string;
+    assignedUsers: {
+      weight: number | null;
+      member: {
+        userId: number;
+      };
+    }[];
+  }[];
+};
+
+async function getQueueAndAttributeWeightData<T extends PartialUser & { priority?: number | null }>(
+  allRRHosts: GetLuckyUserParams<T>["allRRHosts"],
+  routingFormResponseId: number,
+  attributeWithWeights: AttributeWithWeights
+) {
+  // variable to return
+  let averageWeightsHosts: { userId: number; weight: number }[] = [];
+
+  const routingFormResponse = await prisma.app_RoutingForms_FormResponse.findFirst({
+    where: {
+      id: routingFormResponseId,
+    },
+    select: {
+      response: true,
+      form: {
+        select: {
+          routes: true,
+          fields: true,
+        },
+      },
+      chosenRouteId: true,
+    },
+  });
```

Comment:
We could reading the response part to handleNewBooking as we would have to do it there as well for other purposes like supporting responses as variables in Event Name

Reply:
Yes, still wanted to do that. Thanks for pointing that out 🙏